targets/TARGET_Maxim/TARGET_MAX32630/spi_api.c

```diff
@@ -167,19 +168,58 @@ int spi_master_write(spi_t *obj, int value)
     return *req.rx_data;
 }
 
-int spi_master_block_write(spi_t *obj, const char *tx_buffer, int tx_length,
-                           char *rx_buffer, int rx_length, char write_fill) {
-    int total = (tx_length > rx_length) ? tx_length : rx_length;
+//******************************************************************************
+int spi_master_block_write(spi_t *obj, const char *tx_buffer, int tx_length, char *rx_buffer, int rx_length, char write_fill)
+{
+    spim_req_t req;
+
+    if (!(tx_length | rx_length) ||
+        (tx_length < 0) ||
+        (rx_length < 0)) {
+        return 0;
+        }
+
+    req.width = SPIM_WIDTH_1;
+    req.ssel = 0;
+    req.deass = 1;
+    req.callback = NULL;
 
-    for (int i = 0; i < total; i++) {
-        char out = (i < tx_length) ? tx_buffer[i] : write_fill;
-        char in = spi_master_write(obj, out);
-        if (i < rx_length) {
-            rx_buffer[i] = in;
+    core_util_critical_section_enter();
+    if (tx_length == rx_length) {
+        req.tx_data = (uint8_t *)tx_buffer;
+        req.rx_data = (uint8_t *)rx_buffer;
+        req.len = tx_length;
+        SPIM_Trans(obj->spi, &req);
+    } else if (tx_length < rx_length) {
+        req.tx_data = (tx_length > 0) ? (uint8_t *)tx_buffer : NULL;
+        req.rx_data = (uint8_t *)rx_buffer;
+        req.len = (tx_length > 0) ? tx_length : rx_length;
+        SPIM_Trans(obj->spi, &req);
+
+        if (tx_length) {
+            req.tx_data = NULL;
+            req.rx_data = (uint8_t *)(rx_buffer + tx_length);
+            req.len = rx_length - tx_length;
+            SPIM_Trans(obj->spi, &req);
+    }
```

Comment:
can you fix alignment for `}` . this should be 4 spaces right. Line 180 should be 4 spaces right

Reply:
Indentation fixed at line 204.